app/models/miq_provision_request.rb

```diff
@@ -29,6 +29,10 @@ def self.request_task_class_from(attribs)
     vm_or_template = VmOrTemplate.find_by(:id => source_id)
     raise MiqException::MiqProvisionError, "Unable to find source Template/Vm with id [#{source_id}]" if vm_or_template.nil?
 
+    if vm_or_template.ext_management_system.nil?
```

Comment:
I think I would like this better if we pulled both checks (line 30 and line 32) into a single method. Since we check for both a `template` and with this change an `ems` it makes sense to consolidate the code that in either case can `raise` the same exception.